pkg/model/iam/iam_builder.go

```diff
@@ -244,6 +234,41 @@ func (b *IAMPolicyBuilder) BuildAWSIAMPolicy() (*IAMPolicy, error) {
 	return p, nil
 }
 
+// addEC2Permissions updates the IAM Policy with statements granting tailored
+// access to EC2 resources, depending on the instance role
+func addEC2Permissions(p *IAMPolicy, iamPrefix string, b *IAMPolicyBuilder, wildcard stringorslice.StringOrSlice) {
+	if (b.Role == api.InstanceGroupRoleNode) || (b.Role == api.InstanceGroupRoleMaster) {
+		p.Statement = append(p.Statement, &IAMStatement{
+			Effect:   IAMStatementEffectAllow,
+			Action:   stringorslice.Slice([]string{"ec2:Describe*"}),
+			Resource: wildcard,
+		})
+	}
+
+	if b.Role == api.InstanceGroupRoleMaster {
+		p.Statement = append(p.Statement,
+			&IAMStatement{
+				Effect: IAMStatementEffectAllow,
+				Action: stringorslice.Slice([]string{
+					"ec2:ModifyInstanceAttribute",
+					"ec2:CreateRoute",
+				}),
+				Resource: wildcard,
+			},
+			&IAMStatement{
+				Effect:   IAMStatementEffectAllow,
+				Action:   stringorslice.Slice([]string{"ec2:*"}),
+				Resource: wildcard,
+				Condition: Condition{
+					"StringEquals": map[string]string{
+						"ec2:ResourceTag/KubernetesCluster": b.Cluster.GetName(),
```

Comment:
That is a good point... I guess we should continue to honor the existing label for now.  We can either just have two IAM grants (I believe), one on the old and one on the new tags, both with `ec2:*`, or we can try to be granular now.  I'm inclined to to do the former, or to do particular rules for subnets & vpcs (which will soon only have the new tags).  I think technically we tag with both for now, even on VPCs & subnets, so it is OK to do this as-is for now, but we'll have to fix this going forward.